docs/configuration_samples/sample_s3writer/S3Writer.md

```diff
@@ -0,0 +1,20 @@
+# S3 Writer
+
+## Description
+Adding a new LogStreamWriter, S3 Writer, to write logs to S3 by Singer. This involves a new writer configuration as shown below.
+Singer employs a buffer directory to store log messages in buffer files before uploading them to S3. The size of these buffered files is important to take into account for thresholds.
+The current design employs two thresholds, a timer-based (in seconds) and file-sized threshold (in megabytes). The writer will upload the log file to S3 when either of the thresholds is met first.
+For example, if 50MB of log messages is reached before 5 seconds, the writer will upload the log file to S3 because the file size threshold has been exceeded. If 5 seconds is reached before 50MB, the writer will upload the log file to S3 because the timer threshold has expired.
+
+## Example writer configuration in Singer Log Configuration
```

Comment:
nit: Is "Singer Log Configuration" more accurate or "Singer Logstream Configuration"?

Reply:
Ah good catch Vahid, singer Log Stream Configuration is more descriptive. Thank you!